File: sdk/javascript/packages/oracle/src/OciSessionConfig.ts
import { ConfigFileAuthenticationDetailsProvider } from "oci-common";

export class OCISessionConfig {
  ociConfigFileLocation: string;
  profile?: string;
  ksmEndpoint: string;

  constructor(
    ociConfigFileLocation: string,
    profile: string | null,
    kmsEndpoint: string
  ) {
    this.ociConfigFileLocation = ociConfigFileLocation;
    this.profile = profile || "DEFAULT";
    this.ksmEndpoint = kmsEndpoint;
  }

  public getProvider(): ConfigFileAuthenticationDetailsProvider { 
    return new ConfigFileAuthenticationDetailsProvider(this.ociConfigFileLocation, this.profile);
  }

  public getKmsEndpoint(): string {
    return this.ksmEndpoint;
  }
}

File: sdk/python/storage/keeper_secrets_manager_storage/storage_oci_key_management/oci_session_config.py
from oci.config import from_file
from typing import Optional

class OCISessionConfig:
    def __init__(self, oci_config_file_location: str, profile: Optional[str] = None, kms_endpoint: str = ""):
        self.oci_config_file_location = oci_config_file_location
        self.profile = profile if profile else "DEFAULT"
        self.kms_endpoint = kms_endpoint

    def get_provider(self):
        return from_file(self.oci_config_file_location, self.profile)

    def get_kms_endpoint(self) -> str:
        return self.kms_endpoint
